petastorm/spark/spark_dataset_converter.py

```diff
@@ -407,6 +411,45 @@ def _materialize_df(df, parent_cache_dir_url, parquet_row_group_size_bytes,
     return save_to_dir_url
 
 
+def _wait_for_fs_eventually_consistency(url_list):
+    """
+    This is used in databricks runtime. Waiting about 30 seconds to make sure
+    all files can be accessed by dbfs fuse.
+    This is because of s3 eventually consistency, and the backend of dbfs is s3.
+    """
+    fs, path_list = get_filesystem_and_path_or_paths(url_list)
+    remaining_list = list(path_list)
+    new_remaining_list = []
+
+    wait_seconds = _get_spark_session().conf \
+        .get(SparkDatasetConverter.WAIT_FS_EVENTUALLY_CONSISTENCY_SECONS, '0')
+    wait_seconds = int(wait_seconds)
+
+    if wait_seconds <= 0:
+        return
+
+    logger.debug('Because of filesystem eventually consistency, waiting several seconds until '
+                 'these files become accessible: %s', ','.join(remaining_list))
+
+    for _ in range(wait_seconds):
+        for path in remaining_list:
+            if not fs.exists(path):
```

Comment:
minor: it might be better if we use multithread here to check all files in parallel